File: TownOfSalemCalculator/src/townofsalemcalculator/Conditions/Concrete/NightMessages/Witched.java
package townofsalemcalculator.Conditions.Concrete.NightMessages;

import java.util.Arrays;
import java.util.List;
import static townofsalemcalculator.Role.Witch;
import townofsalemcalculator.Conditions.RoleGroup.SingleRoleGroup;

/**
 * The Concrete Condition that you were witched
 * @author Multifacio
 * @version 1.1
 * @since 2017-1-28
 */
public class Witched extends NightMessage {
    
    public Witched() {
        super(new SingleRoleGroup(Witch));
    }
    
    @Override
    public String getCondition() {
        return "You feel a mystical power dominating you. You were controlled by a Witch!";
    }

    @Override
    public List<String> keyWords() {
        List<String> keyWords = Arrays.asList(new String[]{"Feel", "Mystical", "Power", "Dominating", "Controlled", "Witched", "You"});
        keyWords.addAll(Witch.getKeyWords());
        return keyWords;
    }
}

File: src/Conditions/Concrete/Effects/Witched.py
from src.Concepts.Role import Role
from src.Conditions.Concrete.ExistenceCondition import ExistenceCondition
from typing import FrozenSet

class Witched(ExistenceCondition):
    """ The Witched condition is used when during the night you were controlled by a Witch, i.e. you received the
    message: 'You feel a mystical power dominating you. You were controlled by a Witch!' """

    def __init__(self, amnesiacRemembered: FrozenSet[Role] = None):
        super().__init__(frozenset({Role.WITCH}), amnesiacRemembered)